File: Week_01/1.两数之和.cs
/*
 * @lc app=leetcode.cn id=1 lang=csharp
 *
 * [1] 两数之和
 */

// @lc code=start
public class Solution {
    public int[] TwoSum(int[] nums, int target) {

        Dictionary<int, int> dic = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++) {
            
            int anotherNum = target - nums[i];

            if (dic.ContainsKey(anotherNum)) {
                return new int[]{dic[anotherNum], i };
            }
            
            if (!dic.ContainsKey(nums[i])) {
                dic.Add(nums[i], i);
            }

        }

        throw new Exception("No two sum solution");
    }
}
// @lc code=end

File: Week_02/1.两数之和.java
import java.util.HashMap;
import java.util.Map;

/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer,Integer>();
        for (int i = 0; i < nums.length; i++) {
            int sub = target - nums[i];

            if(map.containsKey(sub)) {
                return new int []{map.get(sub), i};
            }
            map.put(nums[i] , i);
        }

        return new int[]{};
    }
}
// @lc code=end

